orttraining/orttraining/core/framework/checkpointing.h

```diff
@@ -39,6 +41,46 @@ common::Status SaveModelCheckpoint(
     const NameMLValMap& runtime_tensors,
     const std::unordered_map<std::string, std::string>& properties);
 
+// opens file descriptor and calls use_fn
+//   use_fn should have this signature: Status use_fn(int file_descriptor)
+template <typename TUseFileFn>
+common::Status WithOpenFile(const PathString& path, bool readonly, TUseFileFn use_fn) {
```

Comment:
Please check if this is already present in one of the ort utils. If we need to add this, can this be part of the checkpointing.cc ? Do we need it here in .h?

Reply:
Moved WithOpenFile and CreateOrtValuesFromTensorProtos into a new file orttraining/orttraining/core/framework/checkpoint_common.h. Those are utilies that is used by checkpoint.cc/checkpointing.cc and checkpoint_test.cc